pkg/controllers/namespace/subnet_share_controller.go

```diff
@@ -122,6 +128,97 @@ func (r *NamespaceReconciler) getExistingSharedSubnetCRs(ctx context.Context, ns
 	return existingSharedSubnets, nil
 }
 
+func (r *NamespaceReconciler) updateDefaultSubnetSetWithSubnets(name string, subnetSetType string, ns string, subnetNames []string) error {
+	ctx := context.TODO()
+	if err := retry.OnError(retry.DefaultRetry, func(err error) bool {
+		return err != nil
+	}, func() error {
+		list := &v1alpha1.SubnetSetList{}
+		label := client.MatchingLabels{
+			servicecommon.LabelDefaultNetwork: subnetSetType,
+		}
+		if err := r.Client.List(context.Background(), list, label, client.InNamespace(ns)); err != nil {
+			return err
+		}
+		if len(list.Items) > 1 {
+			// Should not reach here
+			return fmt.Errorf("multiple default SubnetSet found, subnetSetType: %s, SubnetSets: %v", subnetSetType, list.Items)
+		}
+		if len(list.Items) == 1 {
+			subnetSetCR := &list.Items[0]
+			if len(subnetNames) == 0 {
```

Comment:
if the shared Subnet CR is not created yet, will there any issue?

Reply:
Shared Subnet CR should already be created. updateDefaultSubnetSetWithSpecifiedSubnets is called after processNewSharedSubnets. If there is any error regarding shared Subnet CR creation, it shall fail at processNewSharedSubnets and retry later.